packages/widgetbook/lib/src/fields/date_time_field.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+
+import 'field.dart';
+import 'field_codec.dart';
+import 'field_type.dart';
+
+class DateTimeField extends Field<DateTime> {
+  DateTimeField({
+    required super.name,
+    super.initialValue,
+    super.onChanged,
+    required this.readOnly,
+  }) : super(
+          type: FieldType.dateTime,
+          codec: FieldCodec<DateTime>(
+            toParam: (value) {
+              final string = asString(value);
+              // replace all instances of ':' with '~' to avoid
+              // issues with retrieving the value from the param
+              // in the query group because it is saved in a map
+              return string.replaceAll(':', timeSeparator);
+            },
+            toValue: (param) {
+              if (param == null) return null;
+              final string = param.replaceAll(timeSeparator, ':');
+              return DateTime.tryParse(string);
+            },
+          ),
+        );
+
+  /// The separator used to replace ':' in the [DateTime] string.
+  static const timeSeparator = '~';
+
+  /// The staring [DateTime] value used for the date and time pickers.
+  static final startDateTime = DateTime(0);
+
+  /// The default [DateTime] value used for the date and time pickers
+  /// and the field.
+  static DateTime get defaultDateTime => DateTime.now();
+
+  /// The ending [DateTime] value used for the date and time pickers.
+  static final endDateTime = DateTime.utc(275760, 09, 13);
+
+  /// Whether the field is read only or not.
+  final bool readOnly;
+
+  @override
+  Widget toWidget(BuildContext context, String group, DateTime? value) {
+    final initial = value ?? initialValue;
+
+    final controller = TextEditingController(
+      text: initial == null ? null : _revertSeparator(codec.toParam(initial)),
+    );
+
+    return TextFormField(
+      controller: controller,
```

Comment:
The controller should not be needed if `updateField` is used, the `value` from `toWidget` should have the updated value, which can be used for the `TextFormField`'s `initialValue`.

```suggestion
      initialValue: '${value.year}-${value.month}-${value.day} ${value.hour}:${value.minute}',
```


> 💡 Pro tip:
> Consider creating an extension on `DateTime` that has a method called `toSimpleFormat()` that will return the "yyyy-MM-dd hh:mm" format. The extension method can be used in both `codec` and `initialValue`.

Reply:
The reason I used the controller was to enforce that the value they enter is a date because as I update the field, I also update the controller.

Using initial value allows the user type in anything into the field, but if that is what is expected, I can remove the controller

Reply:
I prefer going with the initial value approach, because defining the `controller` here is like defining a `TextEditingController` inside the widget's `build` method and it will not be that reliable. Most `Field`s should be stateless, and their state is managed from outside.